docs/modules/ROOT/pages/api/erc20.adoc

```diff
@@ -0,0 +1,372 @@
+
+Returns the ticker symbol of the token.
+
+[.contract-item]
+[[decimals]]
+==== `[.contract-item-name]#++decimals++#++(@self: ContractState) → u8++` [.item-kind]#external#
+
+Returns the number of decimals the token uses - e.g. 8 means to divide the token amount by 100000000 to get its user representation.
+
+For example, if `decimals` equals `2`, a balance of `505` tokens should
+be displayed to a user as `5.05` (`505 / 10 ** 2`).
+
+Tokens usually opt for a value of 18, imitating the relationship between
+Ether and Wei. This is the default value returned by this function, unless
+a custom implementation is used.
+
+NOTE: This information is only used for _display_ purposes: it in
+no way affects any of the arithmetic of the contract.
+
+[.contract-item]
+[[total_supply]]
+==== `[.contract-item-name]#++total_supply++#++(@self: ContractState) → u256++` [.item-kind]#external#
+
+See <<IERC20-total_supply,IERC20::total_supply>>.
+
+[.contract-item]
+[[balance_of]]
+==== `[.contract-item-name]#++balance_of++#++(@self: ContractState, account: ContractAddress) → u256++` [.item-kind]#external#
+
+See <<IERC20-balance_of,IERC20::balance_of>>.
+
+[.contract-item]
+[[allowance]]
+==== `[.contract-item-name]#++allowance++#++(@self: ContractState, owner: ContractAddress, spender: ContractAddress) → u256++` [.item-kind]#external#
+
+See <<IERC20-allowance,IERC20::allowance>>.
+
+[.contract-item]
+[[transfer]]
+==== `[.contract-item-name]#++transfer++#++(ref self: ContractState, recipient: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+See <<IERC20-transfer,IERC20::transfer>>.
+
+[.contract-item]
+[[transfer_from]]
+==== `[.contract-item-name]#++transfer_from++#++(ref self: ContractState, sender: ContractAddress, recipient: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+See <<IERC20-transfer_from,IERC20::transfer_from>>.
+
+[.contract-item]
+[[approve]]
+==== `[.contract-item-name]#++approve++#++(ref self: ContractState, spender: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+See <<IERC20-approve,IERC20::approve>>.
+
+[.contract-item]
+[[increase_allowance]]
+==== `[.contract-item-name]#++increase_allowance++#++(ref self: ContractState, spender: ContractAddress, added_value: u256) → bool++` [.item-kind]#external#
+
+Atomically increases the allowance granted to `spender` by the caller.
+Returns `true` on success.
+
+Emits an <<Approval,Approval>> event.
+
+This function panics if:
+
+- The caller is the zero address.
+- `spender` is the zero address.
+
+[.contract-item]
+[[decrease_allowance]]
+==== `[.contract-item-name]#++decrease_allowance++#++(ref self: ContractState, spender: ContractAddress, subtracted_value: u256) → bool++` [.item-kind]#external#
+
+Atomically decreases the allowance granted to `spender` by the caller.
+Returns `true` on success.
+
+Emits an <<Approval,Approval>> event.
+
+This function panics if:
+
+- The caller is the zero address.
+- `spender` is the zero address.
+
```

Comment:
should we add some title or separator between external and internal functions? (and any other set of functions, such as internal, external, maybe constructor/initializers?

Reply:
I think that's a good idea. I added new headers here: https://github.com/OpenZeppelin/cairo-contracts/pull/689/commits/108dfbffa475461c1b6343a081d7b8e1b407c781

Let me know what you think

Reply:
I think is better is we separate the header section, no the ones with the function description:

This looks redundant to me, the external is already in the top-right corner.

<img width="769" alt="Screenshot 2023-09-13 at 15 24 01" src="https://github.com/OpenZeppelin/cairo-contracts/assets/26070224/ce6ff4a2-704b-4d92-a0c6-abc0e14cc70b">


I think something like this would be better, keeping the descriptions as it is.

<img width="752" alt="Screenshot 2023-09-13 at 15 25 15" src="https://github.com/OpenZeppelin/cairo-contracts/assets/26070224/624ee2bf-5e0f-4122-ab8c-bc4e328ce5b4">

Reply:
i like how it looks now, but i also don't think there's any problem with a bit of added redundancy. docsites are reference sites, not books you read in order, and redundancy is often a good idea to maximize the value obtained by reading a small autocontained version.

i think this redundancy is especially useful for navigation, which is usually hard in API docs

Reply:
I don't disagree with Marto's perspective especially for larger contracts a la ERC721. I say let's add them in. We can always change it later

Reply:
Got it, I will update accordingly, maybe in the upgrades PR, to do it in one place if you agree (that PR contains the rest of my doc PRs except overview)

Reply:
Good idea, and merge that after the others. Works for me!

Reply:
I really like how it looks now!

I think it still looks a bit weird that `increase_allowance` and `decrease_allowance` are above the rest, and with a different color. should we add them to their own impl and place them below? i'd prioritize `IERC20` ones

![image](https://github.com/OpenZeppelin/cairo-contracts/assets/1703831/a32c7a89-283e-4c84-9fac-1b400fcf6362)